core/services/ocr2/plugins/ocr2keeper/evm/abi.go

```diff
@@ -45,8 +61,10 @@ func (rp *evmRegistryPackerV2_0) UnpackCheckResult(key types.UpkeepKey, raw stri
 
 	if !upkeepNeeded {
 		result.State = types.NotEligible
-	} else {
-		var ret0 = new(res)
+	}
+	// if NONE we expect the perform data. if TARGET_CHECK_REVERTED we will have the error data in the perform data used for off chain lookup
+	if result.FailureReason == UPKEEP_FAILURE_REASON_NONE || result.FailureReason == UPKEEP_FAILURE_REASON_TARGET_CHECK_REVERTED {
+		var ret0 = new(performDataWrapper)
```

Comment:
this will not always be the case for registry 2.0.0 (which is deployed on testnets) but it will be the case on 2.0.2

Can we add a simple check on len(rawPerformData) > 0 here?

Reply:
https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/automation/2_0/KeeperRegistry2_0.sol#L40-L43